lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -578,13 +587,48 @@ func newSelectHopHintsCfg(invoicesCfg *AddInvoiceConfig) *SelectHopHintsCfg {
 	}
 }
 
+// HopHintInfo contains the channel information required to create a hop hint.
+type HopHintInfo struct {
+	// IsPublic indicates whether a channel is advertised to the network.
+	IsPublic bool
```

Comment:
Besides `IsPublic` we can add `IsActive` here too and then we can remove `IsChannelActive` from the interface.

Reply:
Adding `IsActive` here would change the way we check link liveliness a bit, since rn we check it every time we add a hop hint. I don't think there's much of a loss checking once up front, since selection should happen fairly quickly (and the link can always flap in the background anyway) so changed!